### ISSUE TITLE
[V2] CRI Recovery Enhancement / Bug Fix

### ISSUE THREAD

**Issue by sunpa93** [kubernetes-sigs/azuredisk-csi-driver#1648]
**What happened**:
1. The driver relies on adding `RecoverAnnotation` to force register an UPDATE event for reconciliation to retrigger any operations that failed to complete due to driver restart or upgrade. https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/b084d41779abcfb6a026949dad092d90c314b1ec/pkg/controller/azvolume.go#L550-L551
However, if driver restarts more than once, any CRI that were recovered from previous restart will already have the `RecoverAnnotation`, making the status update operation a noop. Then, no UPDATE event will be registered, and no recovery reconciliation will be triggered for that CRI.
> Therefore, we need to create a unique UUID per recovery to set as a value for the `RecoverAnnotation`.

2. With v2's asynchronous attach feature, `VolumeAttachment.Status.Attached` no longer truly reflects on disk's attach state. As `ControllerPublishVolume` no longer waits for attachment to complete and instead returns an early success, we cannot really tell if the disk is attached to a node or not by looking at `VolumeAttachment.Status.Attached`. And this becomes a problem when we try to recover states for AzVolumeAttachment CRIs on driver restart or upgrade because the driver is clueless on what state the `AzVolumeAttachment.Status.State` should be at. 
> Therefore, we should fetch VMs from cloud-provider-azure and check disks attached to properly recover the `Status.State` field.

**What you expected to happen**:
* CRI Recovery should properly trigger a controller reconciliation with a unique RecoverAnnotation update.
* AzVolumeAttachment CRI Recovery should have a correct `Status.State`.

**How to reproduce it**:

**Anything else we need to know?**:

**Environment**:

- CSI Driver version:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by alice-zheyan-yu** [comment]
For the first one: use unique RecoverAnnotation, it seems other controllers' Recover() also need to use unique annotations.
For example, in node controller:
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/71dc9d2afad507276f69283410f0bd1136dd7047/pkg/controller/node.go#L126

**Comment by alice-zheyan-yu** [comment]
Questions:

1. What's the right way to restart driver?
2. It seems we don't have a controller for azDriverNode. How to reconcile it after we adding the recover annotation?

**Comment by alice-zheyan-yu** [comment]
Questions:

1. I found that we can use function `GetDiskLun()`(see the code link below) to check if attached. Do we have better way to check state from VM?
 https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/990feb50431fd532b31e27f3d2e71540407c3810/vendor/sigs.k8s.io/cloud-provider-azure/pkg/provider/azure_controller_common.go#L200


2. When we fetch VMs from cloud-provider-azure and find out that the disk is actually not attached, the driver is still clueless on what state the AzVolumeAttachment.Status.State should be at. For this case, what target state should we set for `azvolumeattachment`? Can we set it to be `detached` maybe? That's my best guess.
